benchmarks/functional_autograd_benchmark/utils.py

```diff
@@ -0,0 +1,59 @@
+from collections import defaultdict
+
+# Utilities to make nn.Module functional
+def del_attr(obj, names):
+    if len(names) == 1:
+        delattr(obj, names[0])
+    else:
+        del_attr(getattr(obj, names[0]), names[1:])
+def set_attr(obj, names, val):
+    if len(names) == 1:
+        setattr(obj, names[0], val)
+    else:
+        set_attr(getattr(obj, names[0]), names[1:], val)
+
+def make_functional(mod):
+    orig_params = tuple(mod.parameters())
+    # Remove all the parameters in the model
+    names = []
+    for name, p in list(mod.named_parameters()):
+        del_attr(mod, name.split("."))
+        names.append(name)
+
+    # Make params regular Tensors instead of nn.Parameter
+    orig_params = tuple(p.detach().requires_grad_() for p in orig_params)
+    return orig_params, names
+
+def load_weights(mod, names, params):
+    for name, p in zip(names, params):
+        set_attr(mod, name.split("."), p)
```

Comment:
Does this automatically make the param a `nn.Parameter`? (params is a list regular Tensors from what I can tell in `make_functional`)

Reply:
No it does not. We don't actually want them to be Parameters because they can have a grad_fn.
Also we only use this with `autograd.grad()` so having leafs is not important.